daemon/images/service.go

```diff
@@ -38,6 +38,7 @@ type ImageServiceConfig struct {
 	LayerStores               map[string]layer.Store
 	MaxConcurrentDownloads    int
 	MaxConcurrentUploads      int
+	MaxDownloadAttempts       int
```

Comment:
I tried to implement the uint strategy, but it leads to a lot of different (mainly type conversion) errors that need to be solved. Solving this created a big mess in my branch and I had to revert back to your patch. 

I would suggest to use a uint16 if we want to do this. uint8 leads to a max of 256 and this would result in a total 'trying' time of less than 2 days ( sum of all retries*5 sec ) before it timeouts. This would be fine in most situations, but not for the maritime industry. There ships can lose connection for several days. Uint16 provides the possibility to set a value that gives enough time for the ships to get a good internet connection again.

Reply:
Thanks for trying; yes, let's keep it for now. Looks like we would still need some custom validation in the code, so it wouldn't give much improvement